pmd-cpp/src/test/java/net/sourceforge/pmd/cpd/CPPTokenizerTest.java

```diff
@@ -159,6 +159,31 @@ public void testLexicalErrorFilename() throws Exception {
         tokenizer.tokenize(code, new Tokens());
     }
 
+    @Test
+    public void testStringPrefixes() {
+        final String code = "char     a =  '\\x30';         // character, no semantics" + PMD.EOL
+                + "wchar_t  b = L'\\xFFEF';       // wide character, no semantics" + PMD.EOL
+                + "char16_t c = u'\\u00F6';       // 16-bit, assumed UTF16?" + PMD.EOL
```

Comment:
I'm wondering why the prefixes `u` and `U` don't make any problems - since they are not mentioned in the grammar for the token `CHARACTER`. Do we tokenize these correctly?

Reply:
These were indeed not tokenized correctly. Instead, the prefixes were considered separate identifiers.